### PR TITLE
Update Rust smart contract introduction

### PR DESCRIPTION
**Pull request by bucanero**
This PR updates the Rust smart contract introduction, creating a quick-start guide that should guide the user from start to finish.

It's based on the original `Introduction` article, and I have included snippets from other articles so the quick-start guide can be followed without leaving the article.

### CODE REVIEW

File: docs/develop/contracts/rust/intro.md [hunk 1]
```diff
@@ -356,126 +502,202 @@ assert_eq!(1, contract.get_num());
...
+Follow the instructions to set up a simple Rust smart contract with a React front-end.
+Happy coding!
 
-## Troubleshooting
+## Versioning for this article
 
-If `near` commands return an error containing:
-
->Cannot deserialize the contract state.
+At the time of this writing, this example works with the following versions:
```
Comment (thisisjoshford) [inline docs/develop/contracts/rust/intro.md:699]: Love that you added the versioning with this! Nice job :)

**Review by thisisjoshford — CHANGES_REQUESTED** [review]
Left some minor comments / adjustments @bucanero :)

**Review by mikedotexe — APPROVED** [review]

File: docs/develop/contracts/rust/intro.md (hunk 1 above)
Comment (bucanero) [reply docs/develop/contracts/rust/intro.md:699]: Thanks! 😄

File: docs/develop/contracts/rust/intro.md [hunk 2]
```diff
@@ -1,51 +1,190 @@
...
+Smart contracts from NEAR usually have a primary file that holds the code: `./src/lib.rs`.
+This is the conventional filename for a Rust library. Libraries will work great for compiling 
+into WebAssembly and deployed the blockchain.
+
+> **Note:** Once you test, build, and get ready to deploy, a few more files and folders will be added here.
+
+### Editing `Cargo.toml`
+
+Open `Cargo.toml` in your text editor of choice. This file is in the TOML (Tom’s Obvious,
+Minimal Language) format, which is Cargo’s configuration format.
```
[the remark this answers is not in the document]
Comment (bucanero) [reply docs/develop/contracts/rust/intro.md]: I updated the sentence, adding `similar to a package.json file`.

**Review by thisisjoshford — APPROVED** [review]
Awesome job @bucanero !

**Pull request merged**